python_modules/libraries/dagster-components/dagster_components/lib/sling_replication_collection/component.py

```diff
@@ -52,6 +57,10 @@ class SlingReplicationSchema(ResolvableSchema[SlingReplicationSpec]):
         None,
         description="Customizations to the op underlying the Sling replication.",
     )
+    include_metadata: list[SlingMetadataAddons] = Field(
+        ["column_metadata", "row_count"],
```

Comment:
Having to maintain the list of metadata functions on the Fierld, in the `SlingMetadataAddon`, and the conditionals in the `execute` function leaves us with a lot of touch points. But I think that's okay. This implementation feels quite intuitive to me.

If we plan to introduce more of these functions, we might want to consider something more dynamic.

Reply:
agreed-- I expect this might evolve in the future but this will at least allow clean use of the sling component in DOP